P3609.md

```diff
@@ -0,0 +1,571 @@
+
+The other alternative to allow `co_await` switch sechduler is to make it recognize certain expressions passed to it.
+The paper cites the mechanism used by `unifex`:
+```C++
+auto previous = co_await co_continue_on(new_scheduler);
+```
+
+This fragment will allow the user to explicitly jump to a new scheduler, using the `co_await` construct.
+
+
+# Complexity arguments # {#complexity}
+
+## Con: Teachability and aligning with a simple conceptual model ## {#conceptual-model}
+
+With the introduction of `std::execution` ([@P2300R10]) we can divide the execution of an expression in two modes: **sync** and **async**.
+Conceptually we can distinguish between the two modes by looking at the two aspects related to the completion of of the corresponding work: the **time** and the **place** of the completion.
+Thus, we have the following:
+
+execution mode  completion time  completion place
+--------------- ---------------- ------------------------------------------
+**sync**        immediate        same thread
+**async**       delayed          possible different thread (and scheduler)
+
+If we manage to properly teach the C++ users these two modes, then all the other things in `std::execution` and related facilities are easy to understand.
+
+Insofar as the argument goes, everything that deviates from this model is adding more complexity to C++, will confuse users and will lead to incorrect use.
+
+A scheduler affine `std::lazy` would make the `co_await`-ed expressions something that is half-sync, half-async.
+The completion time is delayed (*async*), but the completion place is on the same scheduler, making it *sync*.
+The matter is even more complex, as the place is not fully *sync*, as landing on the same scheduler doesn't imply landing on the same thread.
+
+In other words, a scheduler affine coroutine will make the `co_await`-ed expressions work differently than senders (in subtle ways), and will make the body of the coroutine behave differently than a synchronous function.
+
+In the spirit of the argument, users will get confused by this behavior, not fully understand how things are working, and this will eventually lead to bugs.
+
+
+## Con: Teachability on main uses of `std::lazy` ## {#teachability}
+
+Moreover, considering that one of the purposes of `std::lazy` is to make senders more approachable, the argument also exposes the following teachability problem:
+
+- we teach people "senders are good for asynchrony, making asynchrony safer"
+- we say that "`std::lazy` makes sender more approachable"
+- with scheduler affinity we are telling users "we limit asynchrony, because people find it hard to use it"
+
+The above three points cannot be put together without some contradiction.
+
+Having a teachability problem, we will make the adoption of `std::lazy` and `std::execution` harder, will confuse the users, and, eventually, this will lead to more bugs.
```

Comment:
I don't follow this at all. This reads like an argument from conclusions. Can you give a realistic example of user confusion from isolating schedulers by default, and the resulting bugs?